origin-discovery/src/apollo/resolvers.js

```diff
@@ -57,28 +58,40 @@ const resolvers = {
     },
 
     async offers (root, args, context, info) {
-      const opts = {}
-      opts.buyerAddress = args.buyerAddress
-      opts.listingId = args.listingId
-      const offers = search.Offer.search(opts)
+      const clause = {}
+      if (args.listingId) {
+        clause.listingId = args.listingId
+      }
+      if (args.buyerAddress) {
+        clause.buyerAddress = args.buyerAddress.toLowerCase()
+      }
+      if (args.sellerAddress) {
+        clause.sellerAddress = args.sellerAddress.toLowerCase()
+      }
+      const rows = await db.Offer.findAll({ where: clause })
```

Comment:
This might open the possibility to get all offers in the DB by not specifying `listingId`, `buyerAddress` or `sellerAddress`.  Should we add an explicit check that at least one of the filters is specified (I know we unfortunately can't do that in Apollo schema). Perhaps something like: 
```
if (Object.keys(clause) === 0)
    throw 'At least one of the following filters needs to be specified: listingId, buyerAddress or sellerAddress'
const rows = await db.Offer.findAll({ where: clause })
```

Reply:
Yes, good idea. Added.